google-startup-scripts/usr/lib/systemd/system/google-shutdown-scripts.service

```diff
@@ -0,0 +1,12 @@
+[Unit]
+Description=Google Compute Engine user shutdown scripts
+Before=shutdown.target reboot.target halt.target
+DefaultDependencies=true
+Wants=local-fs.target network-online.target network.target
+
+[Service]
+ExecStart=/usr/share/google/run-shutdown-scripts
+Type=oneshot
+
+[Install]
+WantedBy=shutdown.target reboot.target halt.target
```

Comment:
> were not triggered by calling stop on the instance, or by running sudo shutdown -h now in the VM

That might be because "DefaultDependencies" was set. Also was google-shutdown-scripts.service enabled at that time?